### DISCUSSION TITLE
v7.2.0-preview.6 Release of PowerShell

### DISCUSSION THREAD

**Discussion by daxian-dbw** [PowerShell/PowerShell#15470] [Show and tell]
## [7.2.0-preview.6] - 2021-05-27

**Note:**
1. Known issue about some filesystem cmdlets do not work correctly with long paths, such as `Remove-Item`, `Rename-Item` and `Set-Location`. For details see: https://github.com/PowerShell/PowerShell/issues/15466
2. PSReadLine 2.2.0-beta1 and 2.2.0-beta2 do not work with this preview of PowerShell due to breaking changes in the prediction interface. The upcoming PSReadLine 2.2.0-beta3 release will resolve this. Use PSReadLine 2.1.0 as the temporary workaround.

### Experimental Features

- **[Breaking Change]** Update prediction interface to provide additional feedback to a predictor plugin (#15421)

### Performance

- Avoid collecting logs in buffer if a pipeline execution event is not going to be logged (#15350)
- Avoid allocation in `LanguagePrimitives.UpdateTypeConvertFromTypeTable` (#15168) (Thanks @xtqqczze!)
- Replace `Directory.GetDirectories` with `Directory.EnumerateDirectories` to avoid array allocations (#15167) (Thanks @xtqqczze!)
- Use `List.ConvertAll` instead of `LINQ` (#15140) (Thanks @xtqqczze!)

### General Cmdlet Updates and Fixes

- Use `AllocConsole` before initializing CLR to ensure codepage is correct for WinRM remoting (PowerShell/PowerShell-Native#70) (Thanks @jborean93!)
- Add completions for `#requires` statements (#14596) (Thanks @MartinGC94!)
- Add completions for comment-based help keywords (#15337) (Thanks @MartinGC94!)
- Move cross platform DSC code to a PowerShell engine subsystem (#15127)
- Fix `Minimal` progress view to handle activity that is longer than console width (#15264)
- Handle exception if ConsoleHost tries to set cursor out of bounds because screen buffer changed (#15380)
- Fix `NullReferenceException` in DSC `ClearCache()` (#15373)
- Update `ControlSequenceLength` to handle colon as a virtual terminal parameter separator (#14942)
- Update the summary comment for `StopTranscriptCmdlet.cs` (#15349) (Thanks @dbaileyut!)
- Remove the unusable alias `d` for the `-Directory` parameter from `Get-ChildItem` (#15171) (Thanks @kvprasoon!)
- Fix tab completion for un-localized `about` topics (#15265) (Thanks @MartinGC94!)
- Remove the unneeded SSH stdio handle workaround (#15308)
- Add `LoadAssemblyFromNativeMemory` API to load assemblies from memory in a native PowerShell host (#14652) (Thanks @awakecoding!)
- Re-implement `Remove-Item` OneDrive support (#15260) (Thanks @iSazonov!)
- Kill native processes in pipeline when pipeline is disposed on Unix (#15287)
- Default to MTA on Windows platforms where STA is not supported (#15106)

### Code Cleanup

<details>

<summary>

<p>We thank the following contributors!</p>
<p>@xtqqczze, @powercode, @bcwood</p>

</summary>

<ul>
<li>Enable <code>nullable</code> in some classes (#14185, #14177, #14159, #14191, #14162, #14150, #14156, #14161, #14155, #14163, #14181, #14157, #14151) (Thanks @powercode!)</li>
<li>Annotate <code>ThrowTerminatingError</code> with <code>DoesNotReturn</code> attribute (#15352) (Thanks @powercode!)</li>
<li>Use <code>GetValueOrDefault()</code> for nullable <code>PSLanguageMode</code> (#13849) (Thanks @bcwood!)</li>
<li>Enable <code>SA1008</code>: Opening parenthesis should be spaced correctly (#14242) (Thanks @xtqqczze!)</li>
</ul>

</details>

### Tools

- Add `winget` release script (#15050)

### Tests

- Enable cross-runtime benchmarking to compare different .NET runtimes (#15387) (Thanks @adamsitnik!)
- Add the performance benchmark project for PowerShell performance testing (#15242)

### Build and Packaging Improvements

<details>

<summary>
Update .NET to version <code>v6.0.0-preview.4</code>
</summary>

<ul>
<li>Suppress prompting when uploading the <code>msixbundle</code> package to blob (#15227)</li>
<li>Update to .NET preview 4 SDK (#15452)</li>
<li>Update <code>AppxManifest.xml</code> with newer OS version to allow PowerShell installed from Windows Store to make system-level changes (#15375)</li>
<li>Ensure the build works when <code>PSDesiredStateConfiguration</code> module is pulled in from PSGallery (#15355)</li>
<li>Make sure daily release tag does not change when retrying failures (#15286)</li>
<li>Improve messages and behavior when there's a problem in finding zip files (#15284)</li>
</ul>

</details>

### Documentation and Help Content

- Add documentation comments section to coding guidelines (#14316) (Thanks @xtqqczze!)

[7.2.0-preview.6]: https://github.com/PowerShell/PowerShell/compare/v7.2.0-preview.5...v7.2.0-preview.6

### SHA256 Hashes of the release artifacts

- powershell-7.2.0-preview.6-linux-alpine-x64.tar.gz
  - E77A889A4DFC858B3C415C3DC2945A09EF190AAA1E4BBA86AE631FDDDCD56F42
- powershell-7.2.0-preview.6-linux-arm32.tar.gz
  - 657A5DF09BCA1EA5294C912AC2F98BF340579E0CB296ED018A585667ADBC93B5
- powershell-7.2.0-preview.6-linux-arm64.tar.gz
  - 2370375958F09D0C19EC4FA698BF34282D630E7EFB7C846DE621323B00F123B2
- powershell-7.2.0-preview.6-linux-x64.tar.gz
  - 69DAACCFA497BEA70798F44888F008E3DAB660260FB5FCABC7F95A63E250FB09
- powershell-7.2.0-preview.6-linux-x64-fxdependent.tar.gz
  - FD900383CB74C6B8F4F6907009B78822EE09AB607B2F1479CA4D89D055E4B32B
- powershell-7.2.0-preview.6-osx-x64.pkg
  - FF0604EAE1769AFBF1A39303E4B2C107E833381099017037EBAF02DC6F7C788A
- powershell-7.2.0-preview.6-osx-x64.tar.gz
  - 3E20637FAA53DEB1531E23D623B7003BA53ACC2D7732737F5E45A1D1733675CA
- PowerShell-7.2.0-preview.6-win-arm32.zip
  - A9317BCA1FE626F868B1F7957C83EF1AD97D8B8839B8373027C6BAE240723022
- PowerShell-7.2.0-preview.6-win-arm64.zip
  - 7DFEFFD980D9F8F2728FDF443597F44911B8991E6ADB077FC2665A0486A29B79
- PowerShell-7.2.0-preview.6-win-fxdependent.zip
  - 4C29422A9189DA7DCD98E12676F1D53E11A43FA8F99E299A74CE5D5F581B8FE6
- PowerShell-7.2.0-preview.6-win-fxdependentWinDesktop.zip
  - 076C904097E55A200899BC1ECBEBB5649DA91C4AEE9E9F37A6CEEB70A0C817C8
- PowerShell-7.2.0-preview.6-win-x64.msi
  - 80DEC33663B740BF17371F21662FD891BE61C09F47C11A51404EC8EF9B3D716E
- PowerShell-7.2.0-preview.6-win-x64.zip
  - 8A01D47957CE8C6D07B1F322754BAB030E2EB4FF9DFB95819A46ECC1273455D3
- PowerShell-7.2.0-preview.6-win-x86.msi
  - 2F07FF917429CB4842CD2AD16FB6C96951300B55E32C75271846A7265B27B3DB
- PowerShell-7.2.0-preview.6-win-x86.zip
  - 8289B53CB0B705F98FF3ED79A3DF2F9C5397DB2CFE9C0631E9F42014855FE63E
- powershell-preview_7.2.0-preview.6-1.deb_amd64.deb
  - 5565B2B4E1A9D140225923852EE535F324B5D52600414B1C7A3617999B1A0593
- powershell-preview-7.2.0_preview.6-1.rh.x86_64.rpm
  - 1C942D104C3F1CFAEAA49B01479DD6DBB05CCF2DC166F79C517B188C118BF5EF

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.2.0-preview.6'>v7.2.0-preview.6 Release of PowerShell</a>.</em>

**Comment by iSazonov** [comment]
I don't get v7.2.0-preview.6 git tag again.

**Reply by daxian-dbw** [reply]
The release branch has been merged into GitHub master with a merge commit, you should be able to get the tag by `git fetch --all -p`. If not, try `git fetch --all -p --tags`.

**Reply by iSazonov** [reply]
It should be automatically. This is again an omission in release process. https://github.com/PowerShell/PowerShell/issues/15244#issuecomment-821214166
/cc @TravisEz13

**Reply by TravisEz13** [reply]
I've updated the instruction for [this step](https://github.com/PowerShell/PowerShell/blob/658837323599ab1c7a81fe66fcd43f7420e4402b/tools/releaseBuild/azureDevOps/releasePipeline.yml#L199) to say

```
For preview releases:
   Merge changes to master
   Use a merge commit to make sure the tag is in the history for master

For rc / stable / servicing releases:
    Make sure PRs are opened to master
    Make sure PR to update changelog is created
```

**Reply by TravisEz13** [reply]
@iSazonov This time it was merged to master.  Are you sure you updated from PowerShell/PowerShell master?

![image](https://user-images.githubusercontent.com/10873629/120027166-e48fb900-bfa7-11eb-95b7-7ad39ccd037e.png)

**Reply by iSazonov** [reply]
Now I see the v7.2.0-preview.6 tag. Thanks!

**Reply by RemMai** [reply]
System.TypeLoadException: Could not load type 'System.Management.Automation.Subsystem.PredictionResult' from assembly 'Microsoft.PowerShell.PSReadLine.Polyfiller, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.


this is bug！！！！


Unable to start！！！！


从6升级到7出现bug。PS无法启动！！！！！


⚡英语不好，全是翻译的。⚡

⚡English is not good,Google Translate⚡

**Reply by TravisEz13** [reply]
@RemMai Please update psreadline, using `install-module psreadline -AllowPrerelease -force`

**Comment by potatoqualitee** [comment]
I am unable to find any msix files, even in preview releases of PowerShell. It seems from previous discussions and docs that it should be there. Am I missing something?

Thanks much,
Chrissy
